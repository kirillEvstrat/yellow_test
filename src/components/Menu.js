import React from "react";
import {NavLink} from "react-router-dom";

const Menu = () => {
    return (
        <div className='header-menu'>
            <div className='header-menu-item'>
            <NavLink to='/jogs'>JOGS</NavLink>
        </div>
            <div className='header-menu-item'>
                <NavLink to='/info'>INFO</NavLink>
            </div>
            <div className='header-menu-item'>
                <NavLink to='/contact'>CONTACT US</NavLink>
            </div>
        </div>
    );
};

export default Menu;