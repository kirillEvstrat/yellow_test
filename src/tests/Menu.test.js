import React from 'react';
import Menu from '../components/Menu';
import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";

test('render menu with routing', () => {
    const component = renderer.create(
        <BrowserRouter>
        <Menu />
        </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    //tree.props.onMouseEnter();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    //tree.props.onMouseLeave();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});