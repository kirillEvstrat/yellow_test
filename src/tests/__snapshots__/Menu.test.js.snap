// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="header-menu"
>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/jogs"
      onClick={[Function]}
    >
      JOGS
    </a>
  </div>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/info"
      onClick={[Function]}
    >
      INFO
    </a>
  </div>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/contact"
      onClick={[Function]}
    >
      CONTACT US
    </a>
  </div>
</div>
`;

exports[`Link changes the class when hovered 2`] = `
<div
  className="header-menu"
>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/jogs"
      onClick={[Function]}
    >
      JOGS
    </a>
  </div>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/info"
      onClick={[Function]}
    >
      INFO
    </a>
  </div>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/contact"
      onClick={[Function]}
    >
      CONTACT US
    </a>
  </div>
</div>
`;

exports[`Link changes the class when hovered 3`] = `
<div
  className="header-menu"
>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/jogs"
      onClick={[Function]}
    >
      JOGS
    </a>
  </div>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/info"
      onClick={[Function]}
    >
      INFO
    </a>
  </div>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/contact"
      onClick={[Function]}
    >
      CONTACT US
    </a>
  </div>
</div>
`;

exports[`render menu with routing 1`] = `
<div
  className="header-menu"
>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/jogs"
      onClick={[Function]}
    >
      JOGS
    </a>
  </div>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/info"
      onClick={[Function]}
    >
      INFO
    </a>
  </div>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/contact"
      onClick={[Function]}
    >
      CONTACT US
    </a>
  </div>
</div>
`;

exports[`render menu with routing 2`] = `
<div
  className="header-menu"
>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/jogs"
      onClick={[Function]}
    >
      JOGS
    </a>
  </div>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/info"
      onClick={[Function]}
    >
      INFO
    </a>
  </div>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/contact"
      onClick={[Function]}
    >
      CONTACT US
    </a>
  </div>
</div>
`;

exports[`render menu with routing 3`] = `
<div
  className="header-menu"
>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/jogs"
      onClick={[Function]}
    >
      JOGS
    </a>
  </div>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/info"
      onClick={[Function]}
    >
      INFO
    </a>
  </div>
  <div
    className="header-menu-item"
  >
    <a
      aria-current={null}
      href="/contact"
      onClick={[Function]}
    >
      CONTACT US
    </a>
  </div>
</div>
`;
